src/main/kotlin/com/babcock/http/Router.kt

```diff
@@ -0,0 +1,58 @@
+package com.babcock.http
+
+import java.io.FileInputStream
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+
+
+class Router {
+
+
+    fun handleTarget(request: HttpReq): FileInputStream { // find files instead of hard code, target params
+        val target = getFile(request.requestTarget)
+        val requestTarget = when (target) {
+
+            "not found" -> {
+                ResponseStatus().setStatus(request, HttpStatusCode.CLIENT_ERROR_404_NOT_FOUND)
+                FileInputStream("src/main/resources/web_files/400/404.html")
+            }
+
+            else -> {
+                ResponseStatus().setStatus(request, HttpStatusCode.SUCCESS_200_OK)
+                FileInputStream(target)
+            }
+        }
+
+        return requestTarget
+    }
+
+    fun fileMatcher(path: String): Boolean {
+        val pattern = Regex(".html")
+        return pattern.containsMatchIn(path)
+    }
+
+    fun getFile(target: String): String {
+        val file = target.lowercase().trim()
+        val webRoot = Paths.get("src/main/resources/web_files")
+        val path: Path?
+        val match = fileMatcher(file)
+
+        if (file == "/") {
+            path = Paths.get("src/main/resources/web_files/index.html")
+
+        } else if (match) {
+            path = Paths.get(webRoot.toString(), target)
+        } else {
+            path = Paths.get(webRoot.toString(), "$target.html")
+        }
+        return when (path?.let { Files.exists(it) }) {
+            true -> path.toString()
+            else -> {
+                "not found"
```

Comment:
I like this way to handle paths programmatically. We are limited to only serving `.html` files, but I think this is a good first implementation

Reply:
Thank you, my plan is to make this more robust with something like this: ^([a-zA-Z0-9\s_\\.\-\(\):])+\(html|json|jpg)$

The thought is that usually a user clicks a link to a resource, which should point the the correct file, but if for whatever reason they need to manually type it in, say "/users/index.html" or "/cats/fluffy.jpg" the method can resolve the file name fully written or just using the name. It doesn't search recursively so as long as the directory is correct and that resource exists, they will be served the file.